jenkinsci/jenkins
# [JENKINS-64151] Set a default file size rotation of AsyncPeriodicWork…

… / AsyncAperiodicWork task logs

See [JENKINS-64151](https://issues.jenkins.io/browse/JENKINS-64151).

Be preventive of intensive task logging, that could severely impact disk usage otherwise, by setting a default size rotation. Chose 10 MB as a default as 10 MB seems like already quite a lot for a single periodic task to log in a file in a day. This is also just above what a periodic task that run **every seconds** (that is extremely aggressive) would log by default (about 7MB and 172800 lines):

```
Started at Mon Nov 14 00:00:01 UTC 2022
Finished at Mon Nov 14 00:00:01 UTC 2022. 0ms
[....]
```

### Testing done

* Started Jenkins fresh
* Run the `FingerprintCleanupThread` in script console with `hudson.model.FingerprintCleanupThread.getInstance().run()` (a core PeriodicWork)
* Fill up the file `$JENKINS_HOME/logs/tasks/Fingerprint cleanup.log` to be just a few bytes above the limit

```
% ls -la $JENKINS_HOME/logs/tasks/
total 20608
drwxr-xr-x  3 myuser  staff        96 14 Nov 15:45 .
drwxr-xr-x  3 myuser  staff        96 14 Nov 15:45 ..
-rw-r--r--  1 myuser  staff  10485868 14 Nov 15:53 Fingerprint cleanup.log
```

* Run the `FingerprintCleanupThread` in script console with `hudson.model.FingerprintCleanupThread.getInstance().run()`
* The file should be rotated:

```
% ls -la $JENKINS_HOME/logs/tasks/
total 20616
drwxr-xr-x  4 myuser  staff       128 14 Nov 15:53 .
drwxr-xr-x  3 myuser  staff        96 14 Nov 15:45 ..
-rw-r--r--  1 myuser  staff       227 14 Nov 15:53 Fingerprint cleanup.log
-rw-r--r--  1 myuser  staff  10485868 14 Nov 15:53 Fingerprint cleanup.log.1
```

### Proposed changelog entries

- Set default file size rotation of AsyncPeriodicWork / AsyncAperiodicWork task logs.

### Proposed upgrade guidelines

- Periodic task log files are now rotated based on the file size with a default size of 10 MB. They were previously not rotated based on the file size. To revert to the previous behavior and disable the file size rotation, the system properties [hudson.model.AsyncAperiodicWork.logRotateSize=-1](https://www.jenkins.io/doc/book/managing/system-properties/#hudson-model-asyncaperiodicwork-logrotatesize) and [hudson.model.AsyncPeriodicWork.logRotateSize=-1](https://www.jenkins.io/doc/book/managing/system-properties/#hudson-model-asyncperiodicwork-logrotatesize) can be used.

### Submitter checklist

  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.

### Desired reviewers

NA

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:



<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7376"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
